src/parse/mod.rs

```diff
@@ -0,0 +1,1937 @@
+                body: Vec::new(),
+            });
+        }
+        while let Some(tok) = self.toks.next() {
+            match tok.kind {
+                '{' => break,
+                '#' => {
+                    if let Some(Token { kind: '{', pos }) = self.toks.peek() {
+                        self.span_before = self.span_before.merge(*pos);
+                        self.toks.next();
+                        let interpolation = self.parse_interpolation()?;
+                        params.push_str(&interpolation.node.to_css_string(interpolation.span)?);
+                    } else {
+                        params.push(tok.kind);
+                    }
+                    continue;
+                }
+                '\n' | ' ' | '\t' => {
+                    self.whitespace();
+                    params.push(' ');
+                    continue;
+                }
+                _ => {}
+            }
+            params.push(tok.kind);
+        }
+
+        let raw_body = self.parse()?;
+        let mut rules = Vec::with_capacity(raw_body.len());
+        let mut body = Vec::new();
+
+        for stmt in raw_body {
+            match stmt {
+                Stmt::Style(..) => body.push(stmt),
+                _ => rules.push(stmt),
+            }
+        }
+
+        if !self.super_selectors.last().is_empty() {
+            body = vec![Stmt::RuleSet {
+                selector: self.super_selectors.last().clone(),
+                body,
+                super_selector: Selector::new(),
+            }];
+        }
+
+        body.append(&mut rules);
+
+        Ok(Stmt::UnknownAtRule {
+            name,
+            super_selector: Selector::new(),
+            params: params.trim().to_owned(),
+            body,
+        })
+    }
+
+    fn import(&mut self) -> SassResult<Vec<Stmt>> {
+        self.whitespace();
+        let mut file_name = String::new();
+        let next = match self.toks.next() {
+            Some(v) => v,
+            None => todo!("expected input after @import"),
+        };
+        match next.kind {
+            q @ '"' | q @ '\'' => {
+                file_name.push_str(
+                    &self
+                        .parse_quoted_string(q)?
+                        .node
+                        .unquote()
+                        .to_css_string(self.span_before)?,
+                );
+            }
+            _ => return Err(("Expected string.", next.pos()).into()),
+        }
+        if let Some(t) = self.toks.peek() {
+            if t.kind == ';' {
+                self.toks.next();
+            }
+        }
+
+        self.whitespace();
+
+        let path: &Path = file_name.as_ref();
+
+        let mut rules = Vec::new();
+        let path_buf = if path.is_absolute() {
+            // todo: test for absolute path imports
+            path.into()
+        } else {
+            self.path
+                .parent()
+                .unwrap_or_else(|| Path::new(""))
```

Comment:
```suggestion
                 .unwrap_or_default()
```

Reply:
Unfortunately `Path` does not implement `Default`, so this change cannot be made. I wonder if there's a reason it doesn't..

Reply:
Oh, I didn't know such things exist, maybe people might have different opinion on default for path which is `.`.